### PR TITLE
KAFKA-18315: Fix to Kraft or remove tests associate with Zk Broker config in DynamicBrokerConfigTest, ReplicaManagerTest, DescribeTopicPartitionsRequestHandlerTest, KafkaConfigTest

### PR DESCRIPTION
**Pull request by m1a2st**
Jira: https://issues.apache.org/jira/browse/KAFKA-18315

`TestUtils.createBrokerConfig` put some zk config, We should check all test is testing Kraft mode, and remove zk test 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/server/DynamicBrokerConfigTest.scala [hunk 1]
```diff
@@ -455,54 +455,52 @@ class DynamicBrokerConfigTest {
...
       configWithoutSecret.dynamicConfig.toPersistentProps(dynamicProps, perBrokerConfig = true)
     } catch {
       case _: ConfigException => // expected exception
     }
     val persistedProps = configWithSecret.dynamicConfig.toPersistentProps(dynamicProps, perBrokerConfig = true)
-    assertFalse(persistedProps.getProperty(SaslConfigs.SASL_JAAS_CONFIG).contains("myLoginModule"),
-      "Password not encoded")
-    val decodedProps = configWithSecret.dynamicConfig.fromPersistentProps(persistedProps, perBrokerConfig = true)
-    assertEquals("myLoginModule required;", decodedProps.getProperty(SaslConfigs.SASL_JAAS_CONFIG))
+    assertEquals(password, persistedProps.getProperty(SaslConfigs.SASL_JAAS_CONFIG))
```
Comment (m1a2st) [inline core/src/test/scala/unit/kafka/server/DynamicBrokerConfigTest.scala:467]: In Kraft mode, `PasswordEncoder` doesn't encrypt the password, see 
https://github.com/apache/kafka/blob/22d1ba8265d44738ff81a5031d523a009237aa46/server-common/src/main/java/org/apache/kafka/security/PasswordEncoder.java#L36

File: core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala [hunk 2]
```diff
@@ -258,31 +258,6 @@ class KafkaConfigTest {
     assertTrue(isValidKafkaConfig(props))
   }
 
-  @Test
```
Comment (m1a2st) [inline core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala:260]: Kraft mode doesn't support `control.plane.listener.name` config, so I remove this test

File: core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala [hunk 3]
```diff
@@ -739,106 +714,107 @@ class KafkaConfigTest {
...
-        if (interBrokerVersion.highestSupportedRecordVersion.value >= messageFormatVersion.highestSupportedRecordVersion.value) {
-          val config = buildConfig(interBrokerVersion, messageFormatVersion)
-          assertEquals(interBrokerVersion, config.interBrokerProtocolVersion)
-          if (interBrokerVersion.isAtLeast(IBP_3_0_IV1))
-            assertEquals(IBP_3_0_IV1, config.logMessageFormatVersion)
-          else
-            assertEquals(messageFormatVersion, config.logMessageFormatVersion)
-        } else {
-          assertThrows(classOf[IllegalArgumentException], () => buildConfig(interBrokerVersion, messageFormatVersion))
+        if (interBrokerVersion.isKRaftSupported) {
```
Comment (m1a2st) [inline core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala]: Kraft mode only support min version `IBP_3_0_IV1`, thus we should add check

File: core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala [hunk 4]
```diff
@@ -1530,56 +1502,10 @@ class KafkaConfigTest {
...
-    KafkaConfig.fromProps(props)
-  }
-
-  @Test
-  def testRejectsNegativeNodeIdForZkBasedCaseWithAutoGenDisabled(): Unit = {
-    // -1 is the default for both node.id and broker.id
-    val props = TestUtils.createBrokerConfig(-1, TestUtils.MockZkConnect, port = TestUtils.MockZkPort)
-    val listeners = "PLAINTEXT://A:9092,SSL://B:9093,SASL_SSL://C:9094"
-    props.setProperty(SocketServerConfigs.LISTENERS_CONFIG, listeners)
-    props.setProperty(ServerConfigs.BROKER_ID_GENERATION_ENABLE_CONFIG, "false")
```
Comment (m1a2st) [inline core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala:1563]: remove zk test

File: core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala [hunk 5]
```diff
@@ -1387,13 +1386,6 @@ class ReplicaManagerTest {
     verifyBecomeFollowerWhenLeaderIsUnchangedButMissedLeaderUpdate(new Properties, expectTruncation = false)
   }
 
-  @Test
-  def testBecomeFollowerWhenLeaderIsUnchangedButMissedLeaderUpdateIbp26(): Unit = {
```
Comment (m1a2st) [inline core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:1391]: Kraft doesn't support `IBP_2_6_IV0`, remove it

File: core/src/test/scala/unit/kafka/server/DynamicBrokerConfigTest.scala [hunk 6]
```diff
@@ -450,54 +450,52 @@ class DynamicBrokerConfigTest {
   }
 
   @Test
-  def testPasswordConfigEncryption(): Unit = {
-    val props = TestUtils.createBrokerConfig(0, TestUtils.MockZkConnect, port = 8181)
+  def testPasswordConfigNotEncryption(): Unit = {
+    val props = TestUtils.createBrokerConfig(0, null, port = 8181)
```
Comment (FrankYang0529) [inline core/src/test/scala/unit/kafka/server/DynamicBrokerConfigTest.scala:453]: It looks like following test cases still use `MockZkConnect`. Could you also update it? Thanks.

* testDynamicConfigInitializationWithoutConfigsInZK
* testUpdateRemoteLogManagerDynamicThreadPool
* testRemoteLogDynamicThreadPoolWithInvalidValues

**Review by FrankYang0529 — COMMENTED** [review]
Hi @m1a2st, thanks for the PR. Leave a comment.

File: core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala [hunk 7]
```diff
@@ -733,82 +708,82 @@ class KafkaConfigTest {
...
     assertThrows(classOf[IllegalArgumentException], () => KafkaConfig.fromProps(props))
   }
 
   @Test
   def testEqualAdvertisedListenersProtocol(): Unit = {
-    val props = TestUtils.createBrokerConfig(0, TestUtils.MockZkConnect, port = 8181)
-    props.setProperty(SocketServerConfigs.LISTENERS_CONFIG, "PLAINTEXT://localhost:9092,SSL://localhost:9093")
-    props.setProperty(SocketServerConfigs.ADVERTISED_LISTENERS_CONFIG, "PLAINTEXT://localhost:9092,SSL://localhost:9093")
+    val props = TestUtils.createBrokerConfig(0, null, port = 8181)
+    props.setProperty(SocketServerConfigs.ADVERTISED_LISTENERS_CONFIG, "PLAINTEXT://localhost:9092")
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala]: we should add security map, right?

**Review by chia7712 — COMMENTED** [review]
@m1a2st thanks for this patch. I have added a small comment. PTAL

**Review by FrankYang0529 — APPROVED** [review]
LGTM. Thanks for the patch.

**Review by chia7712 — APPROVED** [review]

**Pull request merged**